daemon/cmd/daemon.go

```diff
@@ -343,25 +345,21 @@ func (d *Daemon) restoreCiliumHostIPs(ipv6 bool, fromK8s net.IP) {
 	}
 
 	restoredIP := node.RestoreHostIPs(ipv6, fromK8s, fromFS, cidrs)
-	if err := removeOldRouterState(ipv6, restoredIP); err != nil {
-		log.WithError(err).Warnf(
-			"Failed to remove old router IPs (restored IP: %s) from cilium_host. Manual intervention is required to remove all other old IPs.",
-			restoredIP,
-		)
-	}
+	return removeOldRouterState(ipv6, restoredIP)
 }
 
 // removeOldRouterState will try to ensure that the only IP assigned to the
 // `cilium_host` interface is the given restored IP. If the given IP is nil,
 // then it attempts to clear all IPs from the interface.
 func removeOldRouterState(ipv6 bool, restoredIP net.IP) error {
 	l, err := netlink.LinkByName(defaults.HostDevice)
-	if errors.As(err, &netlink.LinkNotFoundError{}) && restoredIP == nil {
-		// There's no old state remove as the host device doesn't exist and
-		// there's no restored IP anyway.
-		return nil
-	} else if err != nil {
-		return err
+	if err != nil {
+		if errors.As(err, &netlink.LinkNotFoundError{}) && restoredIP == nil {
+			// There's no old state remove as the host device doesn't exist and
+			// there's no restored IP anyway.
+			return nil
+		}
+		return resiliency.NewRetryableErr(err)
```

Comment:
Good point Timo! I am open to a rename but I think for the time being the naming make sense. 
`Retry()` feels a bit more obscure since we are not performing an actual retry here but wrapping the error to potentially trap it further up the chain. 
In the future, should we opt that additional context around the condition would be deemed useful we could rename accordingly?

Reply:
Ok I'll make it so. Tx Timo!